contracts/upgradeable_contracts/arbitrary_message/BasicHomeAMB.sol

```diff
@@ -38,16 +38,24 @@ contract BasicHomeAMB is BasicAMB, MessageDelivery, MessageProcessor {
         }
     }
 
+    /**
+     * Parses given message, processes a call inside it
+     * @param _message relayed message
+     */
     function handleMessage(bytes _message) internal {
+        bytes32 messageId;
+        uint256 chainId;
         address sender;
         address executor;
-        bytes32 txHash;
-        uint256 gasLimit;
+        uint32 gasLimit;
         bytes1 dataType;
         uint256 gasPrice;
         bytes memory data;
-        (sender, executor, txHash, gasLimit, dataType, gasPrice, data) = ArbitraryMessage.unpackData(_message, false);
-        processMessage(sender, executor, txHash, gasLimit, dataType, gasPrice, data);
+        (messageId, chainId, sender, executor, gasLimit, dataType, gasPrice, data) = ArbitraryMessage.unpackData(
```

Comment:
It seems that it is a good place to check that the chain Id of the contract matches with the chain Id in the message.